### DISCUSSION TITLE
Plotting large regions using local bam file with jbrowse/img

### DISCUSSION THREAD

**Discussion by kcleal** [GMOD/jbrowse-components#3331] [Q&A] [answered]
Im trying to plot a large region view using jbrowse/img. I have been trying this with:

    jb2export --fasta $HG19 --bam chr1.bam --loc chr1:10000-30000 --out jbrowse.png --configtracks bam force:true --config alignments.config

The config file I made was:

    {
      "trackId": "bam",
      "name": "bam",
      "assemblyNames": ["hg19"],
      "type": "AlignmentsTrack",
      "adapter": {
        "type": "BamAdapter",
        "bamLocation": {
          "uri": "chr1.bam"
        },
        "index": {
          "location": {
            "uri": "chr1.bam.bai"
          }
        }
      }
    }

However I get an error message ``Error: Could not resolve identifier "bam"``

How can I fix this? Thanks

**Comment by cmdcolin** [comment]
I might have to look into this, I thought something like force:true just on the --bam flag would work but it throws an error

```
jb2export --fasta $HG19 --bam chr1.bam force:true --loc chr1:10000-30000 --out jbrowse.png
[
  TypeError: self.PileupDisplay.setUserFeatureScreenDensity is not a function
      at Object.setUserFeatureScreenDensity (/home/cdiesh/.fnm/node-versions/v18.12.1/installation/lib/node_modules/@jbrowse/img/node_modules/@jbrowse/plugin-alignments/dist/LinearAlignmentsDisplay/models/model.js:196:32)
      at executeAction (/home/cdiesh/.fnm/node-versions/v18.12.1/installation/lib/node_modules/@jbrowse/img/node_modules/mobx/dist/mobx.cjs.development.js:1257:15)
      at bound setUserFeatureScreenDensity (/home/cdiesh/.fnm/node-versions/v18.12.1/installation/lib/node_modules/@jbrowse/img/node_modules/mobx/dist/mobx.cjs.deve
...
```

I'll check into it and let you know. super impressed by the gw tool, i imagine @jbrowse/img will probably get trounced by it but would be  fun to compare

**Reply by kcleal** [reply]
Hi @cmdcolin, thanks for the quick response - and your nice comment about gw. Im trying to compare as many tools as possible so any help would be much appreciated, thanks!

**Comment by cmdcolin** [comment, marked as answer]
@kcleal i think the force flag should work now, can reinstall with `npm install -g @jbrowse/img` for latest.  i tried plotting some largish region of long reads, 900kb

    jb2export --fasta http://jbrowse.org/genomes/GRCh38/fasta/hg38.prefix.fa.gz --cram http://s3.amazonaws.com/jbrowse.org/genomes/GRCh38/skbr3/skbr3.ont.sort.mod.cram force:true --loc 1:1,000,000-1,900,000

with larger region it is a out of memory.  might need to add node env variables to give it more memory for these cases

the --fasta and --cram/--bam flags should work on local files as well as urls. plotting large e.g. megabase scale regions with alignments is definitely an interesting challenge :) i normally conceive of people creating e.g. a bigwig or some reduced version of the data but it is often interesting to get the raw data too

**Reply by cmdcolin** [reply]
also just for reference, using --configtracks is used for picking specific trackIds out of a config.json (which is normally created by the @jbrowse/cli of the web, and contains assemblies, tracks, etc.)

**Comment by kcleal** [comment]
Thanks so much! I will give it a try on Friday

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________
